adapters/adhese/adhese.go

```diff
@@ -18,267 +14,197 @@ import (
 	"github.com/prebid/prebid-server/v2/openrtb_ext"
 )
 
-type AdheseAdapter struct {
+type adapter struct {
 	endpointTemplate *template.Template
 }
 
-func extractSlotParameter(parameters openrtb_ext.ExtImpAdhese) string {
-	return fmt.Sprintf("/sl%s-%s", url.PathEscape(parameters.Location), url.PathEscape(parameters.Format))
+type ExtTargetsAdhese map[string][]string
+type ExtImpAdheseWrapper map[string]ExtTargetsAdhese
+type wrappedBidExt map[string]map[string]string
+
+func makeSlot(params openrtb_ext.ExtImpAdhese) string {
+	return fmt.Sprintf("%s-%s", params.Location, params.Format)
 }
 
-func extractTargetParameters(parameters openrtb_ext.ExtImpAdhese) string {
-	if len(parameters.Keywords) == 0 {
-		return ""
-	}
-	var parametersAsString = ""
-	var targetParsed map[string]interface{}
-	err := json.Unmarshal(parameters.Keywords, &targetParsed)
-	if err != nil {
-		return ""
+func (a *adapter) MakeRequests(
+	request *openrtb2.BidRequest,
+	requestInfo *adapters.ExtraRequestInfo,
+) (
+	[]*adapters.RequestData,
+	[]error,
+) {
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{
+			Message: "No impression in the bid request",
+		}}
 	}
+	imp := &request.Imp[0]
```

Comment:
@mefjush  based on changes it appears that only first imp from request is considered by adapter. Should update bidder docs to explicitly specify this behavior

Reply:
Done in https://github.com/prebid/prebid.github.io/pull/5618/files